src/main/java/software/amazon/awssdk/crt/http/Http2ClientConnection.java

```diff
@@ -0,0 +1,176 @@
+        http2ClientConnectionSendPing(getNativeHandle(), completionFuture, null);
+        return completionFuture;
+    }
+
+    /**
+     * Send a custom GOAWAY frame.
+     *
+     * Note that the connection automatically attempts to send a GOAWAY during
+     * shutdown (unless a GOAWAY with a valid Last-Stream-ID has already been sent).
+     *
+     * This call can be used to gracefully warn the peer of an impending shutdown
+     * (http2_error=0, allow_more_streams=true), or to customize the final GOAWAY
+     * frame that is sent by this connection.
+     *
+     * The other end may not receive the goaway, if the connection already closed.
+     *
+     * @param Http2ErrorCode   The HTTP/2 error code (RFC-7540 section 7) to send.
+     *                         `enum Http2ErrorCode` lists official codes.
+     * @param allowMoreStreams If true, new peer-initiated streams will continue to
+     *                         be acknowledged and the GOAWAY's Last-Stream-ID will
+     *                         be set to a max value. If false, new peer-initiated
+     *                         streams will be ignored and the GOAWAY's
+     *                         Last-Stream-ID will be set to the latest acknowledged
+     *                         stream.
+     * @param debugData        Optional debug data to send. Size must not exceed
+     *                         16KB. null is acceptable to not include debug data.
+     */
+    public void sendGoAway(final Http2ErrorCode Http2ErrorCode, final boolean allowMoreStreams,
+            final byte[] debugData) {
+        throw new CrtRuntimeException("Unimplemented");
+    }
+
+    public void sendGoAway(final Http2ErrorCode Http2ErrorCode, final boolean allowMoreStreams) {
+        throw new CrtRuntimeException("Unimplemented");
+    }
+
+    /**
+     * Increment the connection's flow-control window to keep data flowing.
+     *
+     * If the connection was created with `manualWindowManagement` set true, the
+     * flow-control window of the connection will shrink as body data is received
+     * for all the streams created on it. (headers, padding, and other metadata do
+     * not affect the window). The initial connection flow-control window is 65,535.
+     * Once the connection's flow-control window reaches to 0, all the streams on
+     * the connection stop receiving any further data.
+     *
+     * If `manualWindowManagement` is false, this call will have no effect. The
+     * connection maintains its flow-control windows such that no back-pressure is
+     * applied and data arrives as fast as possible.
+     *
+     * If you are not connected, this call will have no effect.
+     *
+     * Crashes when the connection is not http2 connection. The limit of the Maximum
+     * Size is 2**31 - 1. If the increment size cause the connection flow window
+     * exceeds the Maximum size, this call will result in the connection lost.
```

Comment:
Can it just throw a validation exception instead of losing the connection?

Reply:
We have an async API under the C level. So, we don't really know the new window size until the thread under the hood invoke. Image a case we receiving data while user updating the window. Or, multiple thread update the windows size at the same time. So, nothing much we can do to prevent this happen before head. From C level, we decided to just close the connection.